samples/nrf9160/modem_shell/src/gnss/gnss_supl_support.c

```diff
@@ -125,8 +110,13 @@ int open_supl_socket(void)
 
 	if (err) {
 		/* Unable to connect, close socket */
-		close(supl_fd);
-		supl_fd = -1;
+		shell_error(shell_global,
+			    "GNSS: Could not connect to SUPL server");
+		if (supl_fd > -1) {
```

Comment:
Doesn't make a difference in terms of functionality but would comparison with `supl_fd != -1` be more accurate as you are setting it by default to -1?

Reply:
The idea was that values which are larger than -1 are valid descriptors, for example -2 is not. But you are right, shouldn't make a difference in functionality either way because socket() should return -1 in case of an error and nothing else.

Reply:
Had to look at different code flows more carefully and it seems `supl_fd > -1` is more accurate. I was thinking there was an error case when we didn't hit this branch and release the socket for setsockopt or connect but I think those failures are handled properly.